### PR TITLE
fix: 649 search screen back and forward browsers actions reacting on typying

### LINKED ISSUE #649: Search screen back and forward browser's actions reacting on typying
### Describe the bug

The back and forward browser's history saves changes while the user types in the search box. 

### Steps/Code to Reproduce

- Go to feeds screen
- Start typing 
- Use the back and forward action and see the screen don't change only change the search parameter in the URL

### Expected Results

The history is only saved when the search action is dispatched.

### Actual Results

Browser history is saved per typed character.

### Screenshots

https://github.com/user-attachments/assets/97cf08d9-6be9-4f7e-8f1d-4900a96f33a2

### Files used

_No response_

### Additional notes

_No response_

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**
Closes #649 
update the URL query params only when the search action is dispatched, rather than with each character typed.

**Expected behavior:** 
search for new york in feeds screen, click back navigation button in the browser, it goes back to the previous page instead of  removing the previous keystroke.

**Testing tips:**

Provide tips, procedures and sample files on how to test the feature.
Testers are invited to follow the tips AND to try anything they deem relevant outside the bounds of the testing tips. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [ ] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by davidgamez** [comment]
The initial search query from the home page is not set in the search query input, but the subsequence text is properly kept in it. The attached screen capture shows this.

https://github.com/user-attachments/assets/4f40c322-aaa7-452d-bcf0-5c7ee6f9c775

**Comment by Alessandro100** [comment]
If we do a search and go back it wont update the search results. This also happens with pagination.

Expected result: If we go back, the data on the page should reflect the current url

https://github.com/user-attachments/assets/e8691603-dc3b-44e9-923c-89e09e264e58

File: web-app/src/app/screens/Feeds/index.tsx [hunk 1]
```diff
@@ -71,11 +71,11 @@ export default function Feed(): React.ReactElement {
   const [selectedFeedTypes, setSelectedFeedTypes] = useState(
     getInitialSelectedFeedTypes(searchParams),
   );
-  const [activeSearch, setActiveSearch] = useState('');
+  const [activeSearch, setActiveSearch] = useState(searchParams.get('q') ?? '');
+  const [searchQuery, setSearchQuery] = useState(activeSearch);
```
Comment (davidgamez) [inline web-app/src/app/screens/Feeds/index.tsx:75]: [suggestion]: Having two state variables with the same "meaning" can be confusing and make it hard to follow the code. If possible, I suggest using a single state variable.

**Comment by qcdyx** [comment]
I forgot to mention that there's a bug where the result page doesn't change when the back navigation button is clicked. This issue also occurs on the main branch. I suggest creating a new issue to track this. @emmambd  I'm currently working on the URL change and using a single state variable. @Alessandro100 @davidgamez

File: web-app/src/app/screens/Feeds/index.tsx (hunk 1 above)
Comment (qcdyx) [reply web-app/src/app/screens/Feeds/index.tsx:75]: I intentionally use a 'duplicate' searchQuery to address the original issue. In line 178, the onChange function only updates the searchQuery, not the activeSearch, which is responsible for triggering an actual search. That's why it removed a keystroke when the back navigation button is clicked.

**Comment by davidgamez** [comment]
> I forgot to mention that there's a bug where the result page doesn't change when the back navigation button is clicked. This issue also occurs on the main branch. I suggest creating a new issue to track this. @emmambd I'm currently working on the URL change and using a single state variable. @Alessandro100 @davidgamez

I checked in QA and the pagination issue is present, let's keep it outside this PR as @qcdyx suggested.

**Comment by qcdyx** [comment]
https://github.com/MobilityData/mobility-feed-api/issues/667 @emmambd

**Review by Alessandro100 — APPROVED** [review]
LGTM, intuitive way of triggering the search ✅
